docs/index.md

```diff
@@ -3992,26 +3998,68 @@ The search pattern language is very simple,
 Note that search will *only* match on the conclusion part
 of an axiom or theorem (there is currently no mechanism to search hypotheses).
 
-A pattern must be a space-separated sequence of one or more items.
-Each item must be a constant or a typecode that is valid in the database:
+A search pattern must be a space-separated sequence of zero or
+more sub-patterns, as defined here:
 
-* A constant pattern only matches the same constant.
-* A typecode pattern matches any variable with that typecode.
+~~~~bnf
+search-pattern ::= sub-pattern*
+sub-pattern ::= ['$'modifiers] MATH-SYMBOL*
+modifiers ::= modifier*
+modifier := '+' | '!' | 'h' | 'a'
+~~~~
+
```

Comment:
It was a mistake in my documentation. I updated it. A search pattern must be a space-separated sequence of **one** or
more sub-patterns. Updated grammar:
```
search-pattern ::= sub-pattern+
sub-pattern ::= ['$'modifiers] MATH-SYMBOL+
modifiers ::= modifier+
modifier := '+' | '!' | 'h' | 'a'
```

Reply:
Ok, I'll use that instead!